### PR TITLE
[#1455] improvement(all) Refactor catalog property mapping to engine property

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Use the new `PropertyConverter` in `catalog-common` to replace the old one.

### Why are the changes needed?

We must provide a uniform property mapping system between Gravitino and the query engine.
Fix: #1455 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Existing test can cover this change.

### CODE REVIEW

File: catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    Map<String, String> engineProperties = new HashMap<>();
+    Map<String, String> gravitinoToTrinoMapping = revsereMap(engineToGravitino());
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      String engineKey = gravitinoToTrinoMapping.get(entry.getKey());
+      if (engineKey != null) {
+        engineProperties.put(engineKey, entry.getValue());
+      }
+    }
+    return engineProperties;
+  }
```
[the remark this answers is not in the document]
Comment (yuqi1129) [reply catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java:53]: OK

File: catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      res.put(entry.getValue(), entry.getKey());
+    }
+
+    return res;
+  }
+
+  /** Convert engine properties to Gravitino properties. */
+  public Map<String, String> fromGravitinoProperties(Map<String, String> properties) {
+    Map<String, String> engineProperties = new HashMap<>();
+    Map<String, String> gravitinoToTrinoMapping = revsereMap(engineToGravitino());
```
[the remark this answers is not in the document]
Comment (yuqi1129) [reply catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java:45]: Fixed

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveMetadataAdapter.java [hunk 3]
```diff
@@ -27,13 +27,13 @@ public HiveMetadataAdapter(
 
   @Override
   public Map<String, Object> toTrinoTableProperties(Map<String, String> properties) {
-    Map<String, String> objectMap = tableConverter.toTrinoProperties(properties);
+    Map<String, String> objectMap = tableConverter.fromGravitinoProperties(properties);
```
[the remark this answers is not in the document]
Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveMetadataAdapter.java:30]: OK, I will rename it.

File: catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    for (Map.Entry<String, Object> entry : properties.entrySet()) {
+      String gravitinoKey = trinoToGravitinoMapping.get(entry.getKey());
+      if (gravitinoKey != null) {
+        PropertyEntry<?> propertyEntry = propertyEntryMap.get(gravitinoKey);
+        if (propertyEntry != null) {
+          // Check value is valid.
+          propertyEntry.decode(entry.getValue().toString());
+        }
+        gravitinoProperties.put(gravitinoKey, entry.getValue());
+      }
```
[the remark this answers is not in the document]
Comment (yuqi1129) [reply catalogs/bundled-catalog/src/main/java/com/datastrato/catalog/property/PropertyConverter.java:70]: I would like the implementor to overwrite it and change the policy regarding null values.

**Review by diqiu50 — APPROVED** [review]
LGMT

**Pull request merged**